fossunited/foss_hackathon/doctype/foss_hackathon_participant/foss_hackathon_participant.py

```diff
@@ -38,11 +42,49 @@ def before_save(self):
         if self.has_value_changed("localhost"):
             self.update_request_status()
 
+    def validate(self):
+        if self.wants_to_attend_locally and not self.localhost:
+            frappe.throw("No LocalHost value provided", frappe.ValidationError)
+
+    def add_to_email_group(self):
+        if not frappe.db.exists(
+            EMAIL_GROUP,
+            {
+                "document_type": HACKATHON,
+                "reference_document": self.hackathon,
+                "group_type": "Event Participants",
+            },
+        ):
```

Comment:
Let's keep this, since we do the same for RSVP and CFP Submissions. 
Also there's the existing 2025 FOSS Hack which can be handled with this. 

We can remove it in a future issue if needed